### PR TITLE
Refactor RoomView

### PR DESCRIPTION
**Pull request by richvdh**
Factor out two components from RoomView, for handling the event timeline:

* MessagePanel, which is basically UI-only

* TimelinePanel, which wraps MessagePanel and handles all the pagination and
  read-receipts guff.

### CODE REVIEW

File: src/components/structures/MessagePanel.js [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+        // true if updates to the event list should cause the scroll panel to
+        // scroll down when we are at the bottom of the window. See ScrollPanel
+        // for more details.
+        stickyBottom: React.PropTypes.bool,
+
+        // callback to determine if a user is the magic freeswitch conference
+        // user. Takes one parameter, which is a user id. Should return true if
+        // the user is the conference user.
+        isConferenceUser: React.PropTypes.func,
```
Comment (dbkr) [inline src/components/structures/MessagePanel.js]: Urgh, I had a funny feeling we'd never see the back of this conferencing ming.

File: src/components/structures/MessagePanel.js [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+var React = require('react');
+var sdk = require('../../index');
+
+/* stateless UI component which builds the event tiles in the room timeline.
+ */
+module.exports = React.createClass({
+    displayName: 'MessagePanel',
+
+    propTypes: {
+        // true to give the component a 'display: hidden' style.
```
Comment (dbkr) [inline src/components/structures/MessagePanel.js]: did you mean display: none?

File: src/components/structures/MessagePanel.js [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+
+    /* check the scroll state and send out pagination requests if necessary.
+     */
+    checkFillState: function() {
+        if (this.refs.scrollPanel) {
+            this.refs.scrollPanel.checkFillState();
+        }
+    },
+
+    render: function() {
```
Comment (dbkr) [inline src/components/structures/MessagePanel.js]: We've loosely been sticking with what seems to be an emerging convention for react method ordering (https://reactjsnews.com/react-style-guide-patterns-i-like/) ie. render goes last, not that this is critical.

File: src/components/structures/MessagePanel.js [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+                if (mxEvSender != this.props.ourUserId) {
+                    var hr;
+                    hr = (
+                        <hr className="mx_RoomView_myReadMarker"
+                            style={{opacity: 1, width: '99%'}} 
+                        />);
+                    readMarkerIndex = ret.length;
+                    ret.push(
+                        <li key="_readupto"
+                                className="mx_RoomView_myReadMarker_container">
```
Comment (dbkr) [inline src/components/structures/MessagePanel.js]: Indenting is a bit odd here?

**Comment by dbkr** [comment]
Otherwise, lgtm.

File: src/components/structures/MessagePanel.js (hunk 2 above)
Comment (richvdh) [inline src/components/structures/MessagePanel.js]: yes I did. thanks.

File: src/components/structures/MessagePanel.js (hunk 3 above)
Comment (richvdh) [inline src/components/structures/MessagePanel.js]: This makes sense. Have reordered.

File: src/components/structures/MessagePanel.js (hunk 4 above)
Comment (richvdh) [inline src/components/structures/MessagePanel.js]: There doesn't seem to be a good way of making indentation work for attributes and nested elements. Have tweaked it anyway.

**Comment by richvdh** [comment]
@dbkr, take another look if you like :)

**Comment by ara4n** [comment]
@dbkr has left the building - suggest merge it.

**Pull request merged**